pkg/asset/agent/configimage/configiso.go

```diff
@@ -0,0 +1,159 @@
+package configimage
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vincent-petithory/dataurl"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/image"
+)
+
+const (
+	configImageFilename = "agentconfig.noarch.iso"
+	configCpioArchive   = "config.gz"
+
+	pullSecretManifest          string = "/etc/assisted/manifests/pull-secret.yaml" //nolint:gosec // not hardcoded credentials
+	nmStateConfigManifest       string = "/etc/assisted/manifests/nmstateconfig.yaml"
+	clusterDeploymentManifest   string = "/etc/assisted/manifests/cluster-deployment.yaml"
+	clusterImageSetManifest     string = "/etc/assisted/manifests/cluster-image-set.yaml"
+	agentClusterInstallManifest string = "/etc/assisted/manifests/agent-cluster-install.yaml"
```

Comment:
Is the sshKey specified by the user on install-config included here?

Reply:
The sshKey is in the infraenv ZTP asset which was not planned on being included in the config ISO as its in the unconfigured ignition https://issues.redhat.com/browse/AGENT-558.

Reply:
But the sshKey is supposed to be specified by the user. I mean, the user won't be able to ssh into the machine without specifying the public ssh key.  So IIUC, it should be apart of the config-image API. At least that what I've understood from this explanation in the [ticket](https://issues.redhat.com/browse/AGENT-558):
`Configuration for the pull secret, image mirror, ssh public key, proxy, additional NTP sources, and rendezvous IP should be passed in as usual.`
Am I missing something? @zaneb

Reply:
There's _three_ places an ssh key can appear. They'll often contain the same key (and in the ABI we populate both from the key given in the install-config), but they don't have to.

1. The ssh key in the agent ISO's authorized_keys - this is used to allow debugging of the installation. This has to be in the ignition. In the factory-installed use case this will either not be possible, or a partner-supplied public key will have to be baked into the disk image.
2. The ssh key in the InfraEnv - this is used to generate a discovery image, but we don't do that in ABI so it is actually ignored.
3.  The ssh key in the AgentClusterInstall - this is used to allow debugging of the _cluster_ after installation.

So the answer to the original question is yes, the ssh key from the install-config will appear here in agent-cluster-install.yaml: https://github.com/openshift/installer/blob/master/pkg/asset/agent/manifests/agentclusterinstall.go#L145